app/scripts/e2e-run-all-flows.sh

```
@@ -1,159 +1,454 @@
+# end. The script exits non-zero if any spec failed.
+#
+# (Previously `set -e` caused the first failure to abort the run and made
+# the terminal appear to crash. `set -uo pipefail` preserves error detection
+# without aborting mid-run.)
 #
 set -uo pipefail
 
 APP_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
-cd "$APP_DIR" || { echo "FATAL: could not cd to $APP_DIR" >&2; exit 1; }
+REPO_DIR="$(cd "$APP_DIR/.." && pwd)"
+cd "$APP_DIR"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)